platform/linux-generic/include/odp_ipsec_internal.h

```diff
@@ -94,6 +94,27 @@ int _odp_ipsec_status_send(odp_queue_t queue,
 struct ipsec_sa_s {
 	odp_atomic_u32_t ODP_ALIGNED_CACHE state;
 
+	/*
+	 * State that gets updated very frequently. Grouped separately
+	 * to avoid false cache line sharing with other data.
+	 */
+	struct ODP_ALIGNED_CACHE {
+		/* Statistics for soft/hard expiration */
+		odp_atomic_u64_t bytes;
+		odp_atomic_u64_t packets;
+
+		union {
+			struct {
+				odp_atomic_u64_t antireplay;
+			} in;
+
+			struct {
+				odp_atomic_u64_t counter; /* for CTR/GCM */
+				odp_atomic_u32_t seq;
+			} out;
+		};
+	} hot;
```

Comment:
Should `ROUNDUP_CACHE_LINE()` be used here to guarantee that the `hot` struct is a integral number of cache lines? Otherwise data following hot may reside in the same cache line, leading to the line sharing issues you're trying to avoid here.

Reply:
I am not totally sure what is the correct way to use the alignment attribute but to my limited understanding the alignment attribute in this patch is a type attribute (as opposed to a variable attribute) that also makes sizeof(hot) a multiple of cache line size. At least in my GCC 7.1 & x86_64 environment that is what happens and struct hot is not sharing cache lines with what follows it.